activerecord/lib/active_record/migration.rb

```diff
@@ -1195,9 +1195,11 @@ def load_migrated
 
       # Used for running a specific migration.
       def run_without_lock
-        migration = migrations.detect { |m| m.version == @target_version }
-        raise UnknownMigrationVersionError.new(@target_version) if migration.nil?
-        result = execute_migration_in_transaction(migration, @direction)
+        if invalid_target?
+          raise UnknownMigrationVersionError.new(@target_version)
+        end
```

Comment:
What does the error message say for the UnknownMigrationVersionError? I'm wondering if it's user friendly enough like "0.1.1" is an unknown migration number. I'm concerned about inexperienced users not realizing their VERSION env var is set elsewhere and how to set it for the Rails migrations.